dashboard/config/script_lessons/20-hour.script_lessons

```diff
@@ -0,0 +1,110 @@
+{
+  "name": "20-hour",
+  "lesson_groups": [
+    {
+      "key": "",
+      "lessons": [
```

Comment:
I think I am missing some of the serializer's magic here. how is it (config/convention) that this file is named `"*.script_lessons"`? 

it looks like there is one .script_lessons file per .script file. if we were to also move .script files onto this format, we could just have one serialized file per script instead of two, correct?

Reply:
That was just the file name I chose when I ran a code snippet to create these files (described in OP):

> Created the new .script_lessons with:
>
>Script.all.each do |s|
  filename = File.join('config', 'script_lessons', "#{s.name}.script_lessons")
  File.write(filename, JSON.pretty_generate(s.serialize_lessons))
end

> If we were to also move .script files onto this format, we could just have one serialized file per script instead of two, correct?

Yea, this was what I was trying to get a prototype working for. It'll just take some more work to even get a prototype working. But I think the payoff could be significant, I think seeding could be a lot faster if we take care to limit the amount of queries needed.

Reply:
Also, Will warned me to not take the idea of redoing script seeding too lightly - he estimated it'd take 2-4 months of work. I still don't think it would take quite that long, but I could see it taking a long time. He thought the approach would work well for new models, though.

Reply:
I find script lessons a little confusing if we aren't tied to that name I would love to change it

Reply:
Any suggestions on name? And is this approach ok in general complexity-wise, since we'd have to keep around the existing `.script` files? @dmcavoy @davidsbailey

Reply:
> Any suggestions on name?

What would make more sense to you, @dmcavoy ? by their nature, each `.script_lessons` file is somewhat redundant with its corresponding `.script` file, because they both contain serialized data pertaining to the script and the objects it owns. However `.script_lessons` is at least somewhat descriptive because everything inside them pertains to lessons within the script, whereas the .script file (mostly) contains attributes of the script itself. 

In terms of script vs unit, I think the order of operations would go (1) rename the Script model to Unit, then (2) rename .script and .script_lessons files to .unit and .unit_lessons. but I think it would be more confusing to have .script files alongside .unit_lessons files in the short term.

> is this approach ok in general complexity-wise, since we'd have to keep around the existing .script files?

yes, I'm generally on board with adding O(100) files to github (assuming you mean big-O complexity I guess). I would still like to understand how feasible it is to create these "on demand", i.e. only create the .script_lessons file when it would actually contain anything. it would help convince me that we're not messing with existing scripts while we test out how the new stuff works. but I'm open to skipping this optimization if we have high confidence that it won't mess up existing scripts as we iterate on the new code.

Reply:
I meant code / system complexity and confusingness. But yea, since most of these new files won't be needed initially, I'm on board with only generating them if they have something we need to store. That would also make it a bit clearer that these are an addition onto the `.script` files. I can work on that after the AMS upgrade.

Reply:
I guess I'm ok with `.script_lessons`. For some reason `.script_lessons` make me think are there lessons not in scripts. However I think thats probably not a big enough reason not to do it and I don't have a better suggestion coming to mind